compiler/compiler/src/main/java/org/robovm/compiler/AppCompiler.java

```diff
@@ -487,23 +494,158 @@ public void failure(Clazz clazz, Throwable t) {
-            return;
+        } else {
+            config.getLogger().info("No classes were modified. Skipping compilation and linking.");
+        }
+    }
+
+    /**
+     * Write the classpaths file that contains a list of class and jar files that were input for the Main binary
+     *
+     * @param classPathsFile
+     * @param linkClasses
+     * @throws IOException
+     */
+    private void storeRecompileInfo(Config config, Set<Clazz> linkClasses) throws IOException {
+        File classPathsFile = new File(config.getTmpDir(), CLASSPATHS_FILENAME);
+        Set<String> paths = new HashSet<>();
+        for (Clazz clazz : linkClasses) {
+            Path path = clazz.getPath();
+            if (path instanceof ZipFilePath) {
+                // comes from a jar - let's consider the jar only.
+                paths.add(path.toString());
+            } else {
+                //is a single class file either in the output dir or in robovm's cache, need to access the real file
+                paths.add(((DirectoryPath.DirectoryPathClazz) clazz).getClassFile().getAbsolutePath());
+            }
         }
 
-        if (linkClasses.contains(config.getClazzes().load(TRUSTED_CERTIFICATE_STORE_CLASS))) {
-            if (config.getCacerts() != null) {
-                config.addResourcesPath(config.getClazzes().createResourcesBootclasspathPath(
-                        config.getHome().getCacertsPath(config.getCacerts())));
+        try (OutputStreamWriter writer = new OutputStreamWriter(new FileOutputStream(classPathsFile), StandardCharsets.UTF_8)) {
+            for (String path : paths) {
+                writer.write(path);
+                writer.write("\n");
             }
         }
 
-        long start = System.currentTimeMillis();
-        linker.link(linkClasses);
-        long duration = System.currentTimeMillis() - start;
-        config.getLogger().info("Linked %d classes in %.2f seconds", linkClasses.size(), duration / 1000.0);
+        // Persisting the config for later comparison
+        File configFile = new File(config.getTmpDir(), "config.xml");
+        try (FileOutputStream output = new FileOutputStream(configFile)) {
+            try {
+                IOUtils.write(configToXml(config), output, StandardCharsets.UTF_8.toString());
+            } catch (Exception e) {
+                config.getLogger().error("Error when computing config's equality. " +
+                        "Forcing recompilation: %s:%s", e.getClass().getSimpleName(), e.getMessage());
+            }
+        }
+
+    }
+
+    /**
+     * Checks, whether recompilation of the Main binary is necessary by looking at the classPathsFile
+     *
+     * @param classPathsFile
+     * @return
+     * @throws IOException
+     */
+    private boolean needsRecompilation(Config config) throws IOException {
+        if (!config.isSmartSkipRebuild()) {
+            config.getLogger().info("Rebuilding because smartSkipRebuild is disabled. Enable it by adding " +
```

Comment:
please remove it as we should not advertise any configuration when it not requested. especially experimental ones.

Reply:
Removed.